Tests/ConfidenceProviderTests/EventSenderEngineTest.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import XCTest
+
+@testable import Confidence
+
+final class MinSizeFlushPolicy: FlushPolicy {
+    private var maxSize = 5
+    private var size = 0
+    func reset() {
+        size = 0
+    }
+    
+    func hit(event: Event) {
+        size += 1
+    }
+    
+    func shouldFlush() -> Bool {
+        return size >= maxSize
+    }
+    
+
+}
+
+final class EventSenderEngineTest: XCTestCase {
```

Comment:
I was thinking about test cases and what do you think about something pseudo like this:
```
// given that engine is initialized with storage that contains 3 events to be uploaded

// when no "sends" happen (in some specific time?)

// expect the 3 events to be uploaded
```
We don't have support for this yet but I think we need it (on both android and ios).

Reply:
you mean introducing flush intervals?

Reply:
yes 👍 but for later.

Reply:
Ticket is created [here](https://spotify.atlassian.net/browse/HAW-531)